### ISSUE TITLE
some vasl setups not linked correctly

### ISSUE THREAD

**Issue by jrvjrv** [vasl-developers/vasl-website#29]
The following vasl setup files are not linked correctly on the download page. Most are likely due to handling of special characters in a script that creates the links. Single quote characters are the largest group. The umlauted ä in Hakkaa Päälle is also an issue, as is the octothorpe (#) in Leibstandarte. I don't know what the problem is for "Norman D"; it may just not be there. This was first reported here: http://www.gamesquad.com/forums/index.php?threads/new-vasl-files-available-at-vasl-info.145998/#post-1916794

BtB2
- BtB-9 Norman D

ESG
- ESG-91 It's Not Over

FrF
- FrF-40 Sporck's Eleven

Leibstandarte
- LSSAH-#2 They Stop Here!

LFT
- FT-168 By Dawn's Early Light

MMP
Core Modules
Hakkaa P??lle!
- ASL-170 11th Company Counterattack
- ASL-172 The Last Attack

General
- G-30 Morgan's Stand

Journal
- J-103 Lenin's Sons
- J-189 Buckley's Block-

StarterKit
- S 022 Another Summer's Day - Enhanced
- S 023 Monty's Gamble - Enhanced
- S 032 Goring's Men - Enhanced
- S 041 Sink's Encouragement - Enhanced

WO
- WO-4 I Don't Like Retreating
- WO-9 Sting 'Em at Zingem

**Comment by zgrose** [comment]
@styson I see you self-assigned right after I made a commit. Testing the changes once it goes live but I think I got them all.

**Comment by zgrose** [comment]
I presume you have some auto-builder going on so this might just be a temp remedy?

**Comment by styson** [comment]
yes, I use a jekyll build process to generate the setup.html file. I will make some changes to that to make the links more reliable based on your commit.

**Comment by jrvjrv** [comment]
You fixed all of them that I tested except BtB-9 Norman D, which is still 404. Also the text for the link for LSSAH-#2 They Stop Here! shows the encoding string instead of an octalthorpe.

**Comment by zgrose** [comment]
Hmm, Norman D downloaded for me in Safari... testing again.

**Comment by zgrose** [comment]
For some reason Safari is showing an extra %20 in the href for the Norman D scenario.

Sam, you'll probably want to tweak your script slightly. Right now you have
````
        downloadPath = path.sub($base, ".").sub("&", "%26").sub("#", "%23")
        $fileHtml.puts '<p><a href="'+downloadPath+'">' + File.basename(downloadPath, ".vsav") + "</a></p>"
````

You don't need to escape the text of the A element in the same way that you escape the href attribute. Something like this might work a bit better:

````
downloadPath = path.sub($base, ".").sub("&", "%26").sub("#", "%23")
displayName = path.sub($base, ".").sub("&", "&amp;").sub("<", "&lt;").sub(">", "&gt;")
$fileHtml.puts '<p><a href="'+downloadPath+'">' + File.basename(displayName, ".vsav") + "
````

If I'm reading it correctly.

**Comment by brookscl** [comment]
Commenting here as I think this is related to what I'm about to work on: updating links to new scenarios I added. The file diff when I generated the new `scen.html` file and the original is fairly significant. Makes me wonder if either my local process isn't operating the same as original / expected, or manual updates have been the norm for a while now. I'm more than happy to work on the script itself if needed and we want to stay on that path.

**Comment by derimmer** [comment]
@brookscl I have just checked vasl.info. The only one of your new files that is showing up is the one which I added to scen.html by hand last night. 

All of your new setup files are now in the vasl-website repo in the setups folder. But your last commit did not update the scen.html file with your new auto-generated version. As you noted, there have been issues with this previously. You and Sam seem to know more about this than me (Jekyll is software and not a book character? LOL) so I am happy to leave it to the two of you to resolve.
